### ISSUE TITLE
Failure in SEAL pip installation

### ISSUE THREAD

**Issue by pranavpawar3** [facebookresearch/SEAL#5]
Hi,

I am facing issues in pip installation.

Initially, due to the git permission issue, I could not clone the repo in a recursive manner so I manually put the SDSL-lite (commit - c32874cb2d8524119f25f3b501526fe692df29f4 ) in res/external dir. 

After that, I installed the necessary requirements, and I have torch 1.10 installed; However, I am not able to install SEAL from the source.

`pip install -e .` gives me the following error - 

```
Obtaining projects/seal-exp/SEAL
Installing collected packages: SEAL
  Running setup.py develop for SEAL
    ERROR: Command errored out with exit status 1:
     command: /home/pranav/.conda/envs/answer_engine_env/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'projects/seal-exp/SEAL/setup.py'"'"'; __file__='"'"projects/seal-exp/SEAL/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: projects/seal-exp/SEAL/
    Complete output (12 lines):
    running develop
    running egg_info
    writing SEAL.egg-info/PKG-INFO
    writing dependency_links to SEAL.egg-info/dependency_links.txt
    writing top-level names to SEAL.egg-info/top_level.txt
    reading manifest file 'SEAL.egg-info/SOURCES.txt'
    writing manifest file 'SEAL.egg-info/SOURCES.txt'
    running build_ext
    building 'seal.cpp_modules._fm_index' extension
    swigging seal/cpp_modules/fm_index.i to seal/cpp_modules/fm_index_wrap.cpp
    swig -python -I../include -c++ -o seal/cpp_modules/fm_index_wrap.cpp seal/cpp_modules/fm_index.i
    error: command 'swig' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/pranav/.conda/envs/answer_engine_env/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'projects/seal-exp/SEAL/setup.py'"'"'; __file__='"'"projects/seal-exp/SEAL/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.

```

Any idea on how to resolve this?

Thanks,
Pranav

**Comment by mbevila** [comment]
What happens if you run SWIG manually?

```
swig -python -I../include -c++ -o seal/cpp_modules/fm_index_wrap.cpp seal/cpp_modules/fm_index.i
```

**Comment by pranavpawar3** [comment]
@mbevila swig wasn't installed in my system. Did it just now, it works fine, able to install SEAL and import the modules.

Thanks.

**Comment by mbevila** [comment]
The permission problem should have been fixed with PR #9. I have also improved the setup guide.
